packages/react-components/react-toast/src/state/useToaster.ts

```diff
@@ -1,24 +1,74 @@
 import * as React from 'react';
-import { useForceUpdate } from '@fluentui/react-utilities';
+import { useEventCallback, useForceUpdate } from '@fluentui/react-utilities';
 import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
-import { Toaster } from './vanilla/toaster';
-import { Toast, ToastPosition, ToasterOptions } from './types';
+import { createToaster } from './vanilla';
+import { Toast, ToastListenerMap, ToastPosition, ToasterId, ToasterOptions } from './types';
 import { ToasterProps } from '../components/Toaster';
+import { EVENTS } from './constants';
 
 export function useToaster<TElement extends HTMLElement>(options: ToasterProps = {}) {
-  const forceRender = useForceUpdate();
+  const forceUpdate = useForceUpdate();
   const toasterOptions = useToasterOptions(options);
-  const [toaster] = React.useState(() => new Toaster());
+  const [toaster] = React.useState(() => createToaster(toasterOptions));
   const { targetDocument } = useFluent();
 
+  const isCorrectToaster = useEventCallback((toasterId: ToasterId | undefined) => {
+    return toasterId === toasterOptions.toasterId;
+  });
+
   React.useEffect(() => {
-    if (targetDocument) {
-      toaster.connectToDOM(targetDocument, toasterOptions);
-      toaster.onUpdate = forceRender;
+    if (!targetDocument) {
+      return;
     }
 
-    return () => toaster.disconnect();
-  }, [toaster, forceRender, toasterOptions, targetDocument]);
+    const buildToast: ToastListenerMap[typeof EVENTS.show] = e => {
+      if (!isCorrectToaster(e.detail.toasterId)) {
+        return;
+      }
```

Comment:
DRY? Can we do something like:

```ts
const buildToast = dispatchForCurrentToaster((e) => {})
```

And hide that check inside `dispatchForCurrentToaster`...

Reply:
yeah I've tried to do a bit better